File: app/src/main/java/com/ogl4jo3/accounting/utility/keyboard/KeyboardUtil.java
package com.ogl4jo3.accounting.utility.keyboard;

import android.app.Activity;
import android.content.Context;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

/**
 * 鍵盤相關工具
 * Created by ogl4jo3 on 2017/6/14.
 */

public class KeyboardUtil {

	/**
	 * 關閉鍵盤
	 *
	 * @param activity Activity
	 * @return 成功關閉
	 */
	public static boolean closeKeyboard(Activity activity) {
		View view = activity.getCurrentFocus();
		if (view != null) {
			((InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE))
			.hideSoftInputFromWindow(view.getWindowToken(), 0);
			return true;
		}
		return false;
	}

	/**
	 * 點按不是EditText的View時，隱藏鍵盤
	 *
	 * @param activity Activity
	 * @param view 需要關閉鍵盤範圍的畫面
	 */
	public static void setupUI(final Activity activity, final View view) {
		// Set up touch listener for non-text box views to hide keyboard.
		if (!(view instanceof EditText)) {
			view.setOnTouchListener(new View.OnTouchListener() {

				public boolean onTouch(View v, MotionEvent event) {
					KeyboardUtil.closeKeyboard(activity);
					return false;
				}
			});
		}

		//If a layout container, iterate over children and seed recursion.
		if (view instanceof ViewGroup) {
			for (int i = 0; i < ((ViewGroup) view).getChildCount(); i++) {
				View innerView = ((ViewGroup) view).getChildAt(i);
				setupUI(activity, innerView);
			}
		}
	}
}

File: app/src/main/kotlin/com/ogl4jo3/accounting/utils/KeyboardUtil.kt
package com.ogl4jo3.accounting.utils

import android.annotation.SuppressLint
import android.app.Activity
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import com.ogl4jo3.accounting.ui.common.extensions.hideKeyboard

/**
 * 鍵盤相關工具
 * Created by ogl4jo3 on 2017/6/14.
 */
object KeyboardUtil {

    /**
     * 點按不是EditText的View時，隱藏鍵盤
     *
     * @param activity Activity
     * @param view     需要關閉鍵盤範圍的畫面
     */
    @SuppressLint("ClickableViewAccessibility")
    fun setupUI(activity: Activity, view: View) {
        // Set up touch listener for non-text box views to hide keyboard.
        if (view !is EditText) {
            view.setOnTouchListener { _, _ ->
                activity.hideKeyboard()
                false
            }
        }

        //If a layout container, iterate over children and seed recursion.
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                val innerView = view.getChildAt(i)
                setupUI(activity, innerView)
            }
        }
    }
}